src/main/kotlin/org/jetbrains/bio/experiments/fit/Span2PeakCallingExperiment.kt

```diff
@@ -0,0 +1,169 @@
+package org.jetbrains.bio.experiments.fit
+
+import org.jetbrains.bio.big.BigWigFile
+import org.jetbrains.bio.coverage.Coverage
+import org.jetbrains.bio.coverage.Fragment
+import org.jetbrains.bio.dataframe.DataFrame
+import org.jetbrains.bio.genome.Chromosome
+import org.jetbrains.bio.genome.ChromosomeRange
+import org.jetbrains.bio.genome.GenomeQuery
+import org.jetbrains.bio.query.CachingQuery
+import org.jetbrains.bio.query.ReadsQuery
+import org.jetbrains.bio.query.reduceIds
+import org.jetbrains.bio.query.stemGz
+import org.jetbrains.bio.statistics.ClassificationModel
+import org.jetbrains.bio.statistics.Fitter
+import org.jetbrains.bio.statistics.hypothesis.NullHypothesis
+import org.jetbrains.bio.statistics.mixture.PoissonRegressionMixture
+import org.jetbrains.bio.statistics.state.ZLH
+import org.jetbrains.bio.util.div
+import org.jetbrains.bio.viktor.asF64Array
+import java.nio.file.Path
+
+class Span2PeakCallingExperiment<Model : ClassificationModel, State : Any>(
+        genomeQuery: GenomeQuery,
+        paths: SpanDataPaths,
+        mapabilityPath: Path?,
```

Comment:
I feel that the overall signature is overloaded and moreover can be possible be splitted into two separate parts - 1. genomeQuery, paths, mappabilityPath, fragment, binSize, unique and part 2: everything else.

Reply:
Also `mappabilityPath` isn't used in SpanPeakCallingExperiment at all, and adding it to `SpanFitInformation` as a hardcoded parameter looks like ownerships inversion. I'd better added some additional information to `Span2FitInformation` superclass of `SpanFitInformation` if this is possible?

Reply:
We were trying to make the two Span models as compatible as possible. One benefit of such approach is that JBR and Span peak-calling module don't care which one they load.
Actually though, since we already know the model type from the file extension, we can make `loadResults` aware of the model type, and thus keep two fit information classes.

Reply:
As for the signature, I concur that it is really heavy. I'll try to split it into smaller parts.

Reply:
It took two days, but I think I've tackled the task. The models should still be interchangeable for JBR and peak-calling, but mapability doesn't leak upstream, and constructor signatures are much lighter. The initialization logic now mostly resides in `getExperiment` methods.